### ISSUE TITLE
azurerm_eventgrid_system_topic is unable to use a management group for the source_arm_resource_id paramater

### ISSUE THREAD

**Issue by searles9** [hashicorp/terraform-provider-azurerm#18034]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.7

### AzureRM Provider Version

3.18.0

### Affected Resource(s)/Data Source(s)

azurerm_eventgrid_system_topic, azurerm_eventgrid_system_topic_event_subscription

### Terraform Configuration Files

```hcl
# Resource group
resource "azurerm_resource_group" "policy_event_grid" {
  name     = "TEST-Policy-Change-EventGrid"
  location = "East US"
}


# Event grid system topic
resource "azurerm_eventgrid_system_topic" "policy_event_topic" {
  name                   = "test-PolicyStateChanges"
  resource_group_name    = azurerm_resource_group.policy_event_grid.name
  location               = "Global"
  source_arm_resource_id = module.management_groups.level_1 # the id of a management group
  topic_type             = "Microsoft.PolicyInsights.PolicyStates"
}
```

### Debug Output/Panic Output

```shell
Error: Can not parse "source_arm_resource_id" as a resource id: No subscription ID found in: "providers/Microsoft.Management/managementGroups/cai-landingzones-level1-mg"
│ 
│   with azurerm_eventgrid_system_topic.policy_event_topic,
│   on vm_agent_install.tf line 47, in resource "azurerm_eventgrid_system_topic" "policy_event_topic":
│   47:   source_arm_resource_id = module.management_groups.level_1
```

### Expected Behaviour

I am trying to get policy state change events into event grid.
I tried to create an azurerm_eventgrid_system_topic.
The source should be a management group (source_arm_resource_id).

This works from the portal and from the CLI, but not from within terraform.
Here is an example CLI call that worked:
az eventgrid system-topic create --name PolicyStateChanges --location global --topic-type Microsoft.PolicyInsights.PolicyStates --source "/tenants/<tenant id>/providers/Microsoft.Management/managementGroups/<mg_name>" --resource-group <rg name> --subscription <subscription id> 

### Actual Behaviour

I get an error that says "Can not parse "source_arm_resource_id" as a resource id: No subscription ID found in: "providers/Microsoft.Management/managementGroups/<mg name>"

It does work if you use a subscription id instead, but I need to make the source a management group. 

I tried two formats for the source_arm_resource_id paramater and both failed:
* providers/Microsoft.Management/managementGroups/<mg name>
* /tenants/<tenat id>/providers/Microsoft.Management/managementGroups/<mg name>"

### Steps to Reproduce

terraform apply

### Important Factoids

Running in an enterprise tenant

**Comment by SWolfeCAI** [comment]
Reproduced and confirmed.  Additionally, when the system topic is created via the Azure Portal and then `terraform import`'ed the same error is also produced -- showing the validation is requiring a subscription id within the resource id, where azure does not include this in this configuration.

**Comment by irishtechie** [comment]
Hi, hope everyone is well.

Is there an update on this bug? I stumbled into it this evening. Going to have to use a local-exec provisioner to accomplish this with Azure CLI.

Thanks in advance.

**Comment by sturlabragason-devoteam** [comment]
As a temporary workaround this should do it:

```hcl
terraform {
  required_providers {
    azapi = {
      source = "Azure/azapi"
    }
  }
}

resource "azapi_resource" "policy_event_topic" {
  type      = "Microsoft.EventGrid/systemTopics@2023-06-01-preview"
  name      = "test-PolicyStateChanges"
  location  = "westeurope"
  parent_id = azurerm_resource_group.policy_event_grid.id
  body = jsonencode({
    properties = {
      source    = module.management_groups.level_1, # the ID of a management group
      topicType = "Microsoft.PolicyInsights.PolicyStates"
    }
  })
}
```

https://registry.terraform.io/providers/Azure/azapi/latest/docs/resources/azapi_resource
https://learn.microsoft.com/en-us/azure/templates/microsoft.eventgrid/systemtopics?pivots=deployment-language-terraform

**Comment by kgopi1** [comment]
Hi, This issue still not resolved with latest azurerm provider (3.87.0) also , still getting validation failed in event system topic when using management group. 

Error message:

 Error: Can not parse "source_arm_resource_id" as a resource id: No subscription ID found in: "tenants/***/providers/Microsoft.Management/managementGroups/******"
│ 
│   with azurerm_eventgrid_system_topic.topic,
│   on ev-policy-events.tf line 19, in resource "azurerm_eventgrid_system_topic" "topic":
│   19:   source_arm_resource_id = "/tenants/***/providers/Microsoft.Management/managementGroups/*****"

Refer the Microsoft link which states to use different syntax when referring the management group id. 

If your Event Grid system topic will be applied to the management group scope, then the Azure CLI --source parameter syntax is a bit different. Here's an example:

az eventgrid system-topic create --name PolicyStateChanges --location global --topic-type Microsoft.PolicyInsights.PolicyStates --source "/tenants/<tenantID>/providers/Microsoft.Management/managementGroups/<management_group_name>" --resource-group "<resource_group_name>"



https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/governance/policy/tutorials/route-state-change-events.md
